assets/js/modules/analytics/datastore/tags.js

```diff
@@ -163,7 +163,14 @@ const baseSelectors = {
 	getExistingTag( state ) {
 		const { existingTag } = state;
 
-		return existingTag;
+		if ( existingTag === undefined ) {
+			return undefined;
+		}
+
+		// It's possible to have an invalid accountID saved by something like the
+		// AMP plugin (see: https://github.com/google/site-kit-wp/issues/1651).
+		// Before returning a tag we should make sure it's valid.
+		return existingTag && isValidPropertyID( existingTag.propertyID ) ? existingTag : null;
```

Comment:
This won't work as-is as `existingTag` _is_ the detected property ID here, not an object 🙂 

```suggestion
		return existingTag && isValidPropertyID( existingTag ) ? existingTag : null;
```

Reply:
Yeah, I caught that but tried this after getting a different error. Must've accidentally pushed this, but I'm still trying to sort out one last test failure 😅